pkg/cache/server/bootstrap/bootstrap.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bootstrap
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/kcp-dev/logicalcluster/v2"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/utils/pointer"
+
+	configcrds "github.com/kcp-dev/kcp/config/crds"
+)
+
+var SystemCRDLogicalCluster = logicalcluster.New("system:system-crds")
+
+func Bootstrap(ctx context.Context, logger logr.Logger, apiExtensionsClusterClient apiextensionsclient.ClusterInterface) error {
```

Comment:
nit: the expectation is that the logger is *in* the context, and you take it out when you need it

Reply:
the logger is modified before calling this function with some contextual data. I don't like the idea of putting it into ctx just to read it here.

Reply:
thoughts?

Reply:
I agree with you in general and I think passing a logger is not that onerous, but k8s has disagreed with both of us and I think conforming to their expectations (least surprise) would likely benefit us